### DISCUSSION TITLE
OPA CPU and Memory Consumption

### DISCUSSION THREAD

**Discussion by birbalekka** [open-policy-agent/community#369] [OPA and Rego]
HI All,
We are running OPA in EKS as an container. While  checking its CPU and memory usages , we see that the cpu doesnt go beyond 1.3 ( allocated 4 ) . Memory - 30 mb ( allocated 2 gb) .
However even with this configuration we are seeing high response times intermittently of upto 13 secs.

Is OPA getting restricted with CPU or memory ?
How can we increase it ?

arguments used during deployment

args:
        - "run"
        - "--ignore=.*"  # exclude hidden dirs created by Kubernetes    
        - "--server"
        - "--tls-cert-file"
        - "{{ .Values.tlsCertFile }}"
        - "--tls-private-key-file"
        - "{{ .Values.tlsPrivateKeyFile }}"
        - "--addr"
        - "{{ .Values.httpsAddr }}"
        - "--addr"
        - "{{ .Values.httpAddr }}"
        - "--log-level"
        - "info"

**Comment by humbertoc-silva** [comment]
Hi @birbalekka,

What about your policies? Are they CPU-bound or IO/bound?
Maybe you are using, for example, the built-int `http.send` to fetch some data to use on policy evaluation and this operation is taking a lot o time. If this is your case, try substituting the `http.send` results with some static in-memory data to see if the latency problem disappears.

**Reply by anderseknert** [reply]
All good questions! In addition to your configuration @birbalekka , I'd suggest adding decision logging to OPA, so that you may get a better idea of what kinds of input would cause such am extreme delay.
